views/js/runtime/qtiCustomInteractionContext.js

```diff
@@ -75,31 +75,14 @@ define(function(){
          * @param {string} pciTypeIdentifier
          * @returns {Object} clonedPciModel
          */
-        createPciInstance : function(pciTypeIdentifier){
+        createPciInstance: function(pciTypeIdentifier) {
+            const registeredPCI = window._pciHooks[pciTypeIdentifier];
 
-            if(window._pciHooks[pciTypeIdentifier]){
-
-                var instance = {},
-                    proto = window._pciHooks[pciTypeIdentifier];
-
-                for(var name in proto){
-                    if(typeof proto[name] === 'function'){
-                        //@todo : delegate function call for better performance ?
-                        instance[name] = proto[name];
-                    }else if(proto[name] !== null && typeof proto[name] === 'object'){
-                        //a plain object:
-                        instance[name] = proto[name].constructor();
-                    }else{
-                        //not an object (nor a function) : e.g. 0, 123, '123', null, undefined
-                        instance[name] = proto[name];
-                    }
-                }
-
-                return instance;
-
-            }else{
```

Comment:
wooh, this looks like we are in the history of JavaScript, like this how our parent were writting code  :rofl:

Reply:
Time is running everything is changed 🤷‍♂️